jose/tests/jws.rs

```diff
@@ -0,0 +1,656 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: Apache-2.0
+
+#![cfg(feature = "jwk")]
+#![cfg(feature = "jws")]
+#![cfg(feature = "rsa")]
+#![cfg(feature = "p256")]
+#![cfg(feature = "hmac")]
+#![cfg(feature = "sha2")]
+
+use jose::alg::Signing::*;
+use jose::b64::Update;
+use jose::jws::crypto::rcrypto::{Signing, Verifying};
+use jose::jws::crypto::{Signer, SigningKey, Verifier, VerifyingKey};
+use jose::jws::Jws;
+use jose::key::rcrypto::{Error, Key};
+
+const EMPTY: [Key<Verifying>; 0] = [];
+
+fn ver<K: Into<Key<Verifying>>, const N: usize>(jws: &Jws, keys: [K; N]) -> Result<(), Error> {
+    let pay = match jws {
+        Jws::Flattened(f) => f.payload.as_ref().unwrap(),
+        Jws::General(g) => g.payload.as_ref().unwrap(),
+    };
+
+    keys.into_iter()
+        .map(|x| x.into())
+        .collect::<Vec<_>>()
+        .verify(jws)
+        .unwrap()
+        .chain(pay)
+        .unwrap()
+        .finish()
+}
+
+fn valid<K: Into<Key<Verifying>>, const N: usize>(jws: &Jws, keys: [K; N]) {
+    ver(jws, keys).unwrap();
+}
+
+fn invld<K: Into<Key<Verifying>>, const N: usize>(jws: &Jws, keys: [K; N]) {
```

Comment:
It’s certainly intentional, but I tend to prefer valid English for clarity. What do you think?

```suggestion
fn invalid<K: Into<Key<Verifying>>, const N: usize>(jws: &Jws, keys: [K; N]) {
```

Reply:
It was intentional. See how this function is used in tests. I had a hard time remembering all the valid vs invalid key combinations. Using `valid()` and `invld()` caused both key sets to have the same horizontal width which aided in my reasoning. Happy to change this though.